### ISSUE TITLE
Unable to upload to testflight using Xcode 14 beta

### ISSUE THREAD

**Issue by ptxmac** [fastlane/fastlane#20371]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When trying to use `upload_to_testflight` with XCode 14 BEta, fastlane reports the following error:
```
[!] Could not find transporter at /Applications/Xcode-beta.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.
```

##### Command executed
`bundle exec fastlane run upload_to_testflight`

##### Relevant output when running fastlane, including the stack trace and command used

<details>
  <pre> 

[15:58:37]: Ready to upload new build to TestFlight (App: xxxx)...
Could not find transporter at /Applications/Xcode-beta.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.
[15:58:38]: Going to upload updated app to App Store Connect
[15:58:38]: This might take a few minutes. Please don't interrupt the script.

[!] Could not find transporter at /Applications/Xcode-beta.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.

</pre>
</details>

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 13.0                                             |
| Ruby                        | 3.1.1                                            |
| Bundler?                    | true                                             |
| Git                         | git version 2.32.1 (Apple Git-133)               |
| Installation Source         | ~/.rbenv/versions/3.1.1/bin/fastlane             |
| Host                        | macOS 13.0 (22A5266r)                            |
| Ruby Lib Dir                | ~/.rbenv/versions/3.1.1/lib                      |
| OpenSSL Version             | OpenSSL 1.1.1n  15 Mar 2022                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/ |
| Xcode Version               | 14.0                                             |
| Swift Version               | 5.7                                              |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.206.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.3.0        |
| did_you_mean                    | 1.6.1        |
| bundler                         | 2.2.31       |
| pathname                        | 0.2.0        |
| rake                            | 13.0.6       |
| rexml                           | 3.2.5        |
| CFPropertyList                  | 3.0.5        |
| public_suffix                   | 4.0.7        |
| addressable                     | 2.8.0        |
| artifactory                     | 3.0.15       |
| atomos                          | 0.1.3        |
| aws-eventstream                 | 1.2.0        |
| aws-partitions                  | 1.598.0      |
| aws-sigv4                       | 1.5.0        |
| jmespath                        | 1.6.1        |
| aws-sdk-core                    | 3.131.1      |
| aws-sdk-kms                     | 1.57.0       |
| aws-sdk-s3                      | 1.114.0      |
| babosa                          | 1.0.4        |
| claide                          | 1.1.0        |
| colored                         | 1.2          |
| colored2                        | 3.1.2        |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| declarative                     | 0.0.20       |
| digest-crc                      | 0.6.4        |
| unf_ext                         | 0.0.8.2      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| dotenv                          | 2.7.6        |
| emoji_regex                     | 3.2.3        |
| excon                           | 0.92.3       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| multipart-post                  | 2.0.0        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.0       |
| http-cookie                     | 1.0.5        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.2.0        |
| fastimage                       | 2.2.6        |
| gh_inspector                    | 1.1.3        |
| jwt                             | 2.4.1        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| signet                          | 0.16.1       |
| googleauth                      | 1.1.3        |
| httpclient                      | 2.8.3        |
| mini_mime                       | 1.1.2        |
| trailblazer-option              | 0.1.2        |
| uber                            | 0.1.0        |
| representable                   | 3.2.0        |
| retriable                       | 3.1.2        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.5.0        |
| google-apis-androidpublisher_v3 | 0.21.0       |
| google-apis-playcustomapp_v1    | 0.7.0        |
| google-apis-iamcredentials_v1   | 0.10.0       |
| google-apis-storage_v1          | 0.14.0       |
| google-cloud-env                | 1.6.0        |
| google-cloud-errors             | 1.2.0        |
| google-cloud-core               | 1.6.0        |
| google-cloud-storage            | 1.36.2       |
| json                            | 2.6.2        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.1.1        |
| plist                           | 3.6.0        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| simctl                          | 1.6.8        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.8.0        |
| terminal-table                  | 1.8.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.21.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
</details>

*generated on:* **2022-06-09**
</details>

**Comment by jbmorley** [comment]
I'm seeing the exact same issue with Xcode 14 beta and macOS Ventura.

**Comment by jbmorley** [comment]
> I'm seeing the exact same issue with Xcode 14 beta and macOS Ventura.

So it looks like Transporter.app is no longer bundled with Xcode 14. I managed to make some pretty good progress with this by installing Transporter.app from the App Store and specifying:

```
FASTLANE_ITUNES_TRANSPORTER_PATH=/Applications/Transporter.app/Contents/itms
```

Disappointingly this now means we hit another error:

```
[13:37:14]: [iTMSTransporter] -Djava.ext.dirs=/Applications/Transporter.app/Contents/itms/java/lib/ext is not supported.  Use -classpath instead.
[13:37:14]: [iTMSTransporter] Error: Could not create the Java Virtual Machine.
[13:37:14]: [iTMSTransporter] Error: A fatal exception has occurred. Program will exit.
```

Curiously https://github.com/fastlane/fastlane/pull/16689 looks like it actually tried to address this lack of support for `-Djava.ext.dirs=` in Transporter.app's bundled `java`. That code is still in `master` (see https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L280), so I wonder if perhaps Xcode 14 and Ventura are causing it to fail to use this more up-to-date code path?

**Comment by doncatas** [comment]
Im facing the same problem. Any news how to solve or we need to downgrade macOS? for using v13 xcode version

**Comment by ptxmac** [comment]
A workaround is to keep XCode 13 installed and use either `xcode-select` to change the default xcode, or overwrite it in the fastlane file (the method is also called `xcode-select` in fastlane.

Xcode 13 won't open on macOS 13, but the cli tools still work perfectly.

Also note that Apple won't even approve Xcode 14 ipa's for the appstore, so having a Xcode 13 install is a must for anyone wishing to release.

**Comment by YahiaRagae** [comment]
Thanks for @ptxmac 's idea 
````
- task: Bash@3
  displayName: 'Release to App store'
  inputs:
    targetType: inline
    script: |
      xcrun altool --validate-app -f my-app.ipa --apiKey $apiKeyId --apiIssuer $apiIssuer --output-format json --type ios
      xcrun altool --upload-app -f my-app.ipa --primary-bundle-id $bundleId --apiKey $apiKeyId --apiIssuer $apiIssuer --type ios
````

But you must ensure that you have installed the API key `P8` file at `~/private_keys` on your machine. and clean it after finishing the build

**Comment by pyby** [comment]
Using Xcode 13 fallback or `altool`, mentioned by @ptxmac, we have an ITMS error with Xcode 14 beta 3. Apple does not allow to upload builds from the current iOS and tvOS beta versions.

`ERROR ITMS-90744: Unsupported Xcode or SDK Version. Your app was built with a version of Xcode or SDK that is not yet supported for upload to App Store Connect. For more information about supported versions of Xcode and SDK for Testflight or App Store submissions, view the App Store Connect What's New page (https://developer.apple.com/app-store-connect/whats-new/).`

Using `FASTLANE_ITUNES_TRANSPORTER_PATH=/Applications/Transporter.app/Contents/itms` on macOS Monterey 12.4 has the same "Java Virtual Machine" issue, mentioned by @jbmorley .

**Comment by EmDee** [comment]
> After few days, and maybe the public betas, ITMS now accept builds built with Xcode 14 beta 3. ✅

Sorry for hijacking this thread, but does Apple already accepts Xcode 14/iOS 16 builds for TestFlight? I can't seem to find any information here.

**Comment by pyby** [comment]
> > After few days, and maybe the public betas, ITMS now accept builds built with Xcode 14 beta 3. ✅
> 
> 
> 
> Sorry for hijacking this thread, but does Apple already accepts Xcode 14/iOS 16 builds for TestFlight? I can't seem to find any information here.

Yes. After few rejected builds when the new Xcode beta was available, now, it works like a charm.

**Comment by adrianwitaszak** [comment]
Not working for me too. Tried Xcode 14 beta 3 on a new project. Any workaround for this?

**Comment by EmDee** [comment]
> Not working for me too. Tried Xcode 14 beta 3 on a new project. Any workaround for this?

What exactly is not working for you?

Have you tried `xcode-select` as mentioned in https://github.com/fastlane/fastlane/issues/20371#issuecomment-1160168429?

Or `altool` as mentioned in https://github.com/fastlane/fastlane/issues/20371#issuecomment-1153737391?

I just successfully uploaded using Xcode 13 (via `xcode-select`).

**Comment by freddi-kit** [comment]
Hello, I did some workaround and investigation. I wrote about them

## Workaround
- Before calling `pilot` action, I called xcode-select("/Application/Xcode13.4.1.app") and it works. 
   - NOTE:  Please replace "/Application/Xcode13.4.1.app" with your env's Xcode 13

## Investigation
- It looks like ITMSTransporter has problem in Xcode 14 beta and now I sent feed back to apple, please wait for answer
   - If you get error message `shell An error (-10814) occurred. The operation couldn’t be completed. (OSStatus error -10814.) (null)`, you looks select xcode version correctly. But please consider to do xcode-select correctly if you got another error message, it should be other issue which I mention.

- The fastlane bahavior with `FASTLANE_ITUNES_TRANSPORTER_PATH` may be bug. I'm not sure why fastlane calls to java runtime directly which is contained in itsm.  This should be done by frontend part of ITMSTransporter. I'm checking how to improve it.

**Comment by adamtharani** [comment]
Any updates on an official fix?

**Comment by mathaeus** [comment]
@joshdholtz sorry to tag you, are you aware of this? :)

**Comment by joshdholtz** [comment]
Oh, I was not aware of this! I totally missed this issue 😱

Will add to my queue!

**Comment by freddi-kit** [comment]
@joshdholtz i'm curious why fastlane call java runtime inside itsm instead of `/Applications/Transporter.app/Contents/itms/bin/iTMSTransporter`. Do you know why?
https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L300

**Comment by freddi-kit** [comment]
Currently i'm working for fixing it

**Comment by freddi-kit** [comment]
Hello. I've got answer from Apple via Feedback. ITMSTransporter is officially removed from Xcode 14 beta and now it is only available on Transporter.app. Or apple recommends to use atool instread.

**Comment by freddi-kit** [comment]
I success to fix it when using iTMSTransporter.app. please wait a moment to open PR

**Comment by freddi-kit** [comment]
I'm sorry, I found `FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT` to make success upload ipa to AppStore with Transporter.app

```
export FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1 export 
export FASTLANE_ITUNES_TRANSPORTER_PATH=/Applications/Transporter.app/Contents/itms 
bundle exec fastlane {your_action}
```

Please set FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT if you want to use Transporter.app.

**Comment by freddi-kit** [comment]
But I'm also working on replacing iTMSTransporter with altool for xcode14 beta user, please wait a moment

**Comment by justincbeck** [comment]
@freddi-kit - Nice job, that worked for me using Xcode 14 RC on MacOS Ventura Beta 6.

**Comment by m-keller** [comment]
After reading all the great answers here, one way that worked for us until a new method is default:

* Extract itms/ folder from XC 13. Can be zipped and moved around between machines
* Set `export FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1` and `export FASTLANE_ITUNES_TRANSPORTER_PATH="/path/to/unzipped/itms`
* Run Fastlane as before against XC 14 and don't get into any issues with ASC

**Comment by everuribe** [comment]
Yikes any chance we can get #20631 in a hot fix to unblock CI builds with Xcode 14. iOS 16 release is in 3 days 😱

**Comment by EmDee** [comment]
Apple still doesn't accept Xcode 14 submissions, no? At least I haven't seen any Apple news in that regard yet.

Of course that's only a matter of hours/days probably.

**Comment by m-keller** [comment]
> Apple still doesn't accept Xcode 14 submissions, no? At least I haven't seen any Apple news in that regard yet.
> 
> Of course that's only a matter of hours/days probably.

Submission is open since September 7: https://developer.apple.com/news/releases/?id=09072022h

**Comment by crazymanish** [comment]
Dear all,

@freddi-kit made a good-progress in this PR https://github.com/fastlane/fastlane/pull/20631
I am adding here the testing steps to test this PR and provide valuable feedback. 

### Testing Steps
- Update `Gemfile` to 👇 and run `bundle install`, `bundle update fastlane`, or `bundle update`

```rb
gem "fastlane", :git => "https://github.com/freddi-kit/fastlane.git", :branch => "altool-upload"
```

- Try to test for all possible combinations
  - [x] `Pilot` using AppStore-connect API key 
  - [x] `Pilot` using username-password (old fashion apple-id, password)
  - [x] `Deliver` using AppStore-connect API key 
  - [x] `Deliver` using username-password (old fashion apple-id, password)

**Comment by EmDee** [comment]
> > Apple still doesn't accept Xcode 14 submissions, no? At least I haven't seen any Apple news in that regard yet.
> > Of course that's only a matter of hours/days probably.
> 
> Submission is open since September 7: https://developer.apple.com/news/releases/?id=09072022h

Oh wow, thanks! Apple usually post it in the [Latest News](https://developer.apple.com/news/) section that's why I've missed it.

**Comment by IanHoar** [comment]
@crazymanish I think you are seeing the same issue as he but I just attempted a deploy with your instructions using Pilot+APIKey and got this: `'build_upload_command': \e[31m[!] undefined method 'shellescape' for nil:NilClass\e[0m (NoMethodError)`

**Comment by crazymanish** [comment]
> @crazymanish I think you are seeing the same issue as he but I just attempted a deploy with your instructions using Pilot+APIKey and got this: 'build_upload_command': \e[31m[!] undefined method 'shellescape' for nil:NilClass\e[0m (NoMethodError)

Indeed! same issue that i m getting. I am just working on it. i will see if i have permission to add commit in the @freddi-kit PR. otherwise i will open PR on top of that PR... i will keep you all posted. 

Thank you for testing!

**Comment by freddi-kit** [comment]
@IanHoar 
I fixed my code, please try it again

**Comment by wrutkowski** [comment]
Can confirm, Apple is accepting submissions of app built with Xcode 14 and uploaded with @freddi-kit workaround from https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663
My app is now in `Pending Apple Release` state.

**Comment by IanHoar** [comment]
@freddi-kit, confirmed working, thanks!

**Comment by crazymanish** [comment]
Dear all,

@freddi-kit PR https://github.com/fastlane/fastlane/pull/20631 looks good now...we have covered all the possible known scenarios.
I am adding `status: workaround available` in this ticket because people are no longer blocked to start using Xcode14.
Now Xcode14 users can switch the Fastlane version to PR https://github.com/fastlane/fastlane/pull/20631 in **local and CI** both and everything should work without any hack and without any change inside the Fastfile-scripts/pilot/deliver code. 


### Switch your Fastlane version to PR
- Update `Gemfile` to 👇 and run `bundle install`, `bundle update fastlane`, or `bundle update`

```rb
gem "fastlane", :git => "https://github.com/freddi-kit/fastlane.git", :branch => "altool-upload"
```

Please let us know if we are still missing any edge-case or something! To me, PR https://github.com/fastlane/fastlane/pull/20631 is simply fire 🔥🔥🔥 good to go 🚀 
Thank you again @freddi-kit 👏

**Comment by drewolbrich** [comment]
Thank you @freddi-kit and everyone else.

**Comment by Roberdan** [comment]
with final version of Xcode14 I'm still having the issue despite the workaround, anyone else?

**Comment by rmdvector** [comment]
not working for me either. xcode must have auto-updated today.

**Comment by hervenivon** [comment]
@Roberdan @rmdvector this workaround https://github.com/fastlane/fastlane/issues/20371#issuecomment-1242768042 worked for me except for the end of processing. `faslane` has never been able to get the update "signal" to finish.

**Comment by rmdvector** [comment]
@hervenivon that was the first thing I tried, didn't work. I'm currently unpacking XCode 13 to see if that works. Meantime I was able to use xcode to upload the archive to testflight.

**Comment by freddi-kit** [comment]
> @Roberdan @rmdvector this workaround https://github.com/fastlane/fastlane/issues/20371#issuecomment-1242768042 worked for me except for the end of processing. faslane has never been able to get the update "signal" to finish.

Me too, and AppStore Connect still show app is in processing so I think it is app store connect side issue

**Comment by freddi-kit** [comment]
Ah, it looks take so long time even if connect side finished processing in my env, i think it can be skipped by `skip_waiting_for_build_processing`. Let me check later but I think it is issue of App side API because there are update of AppStore Connect API today.

**Comment by freddi-kit** [comment]
@hervenivon I investigated and it should be other issue due to API issue, let me check after the this issue and PR is merged, please consider to use `skip_waiting_for_build_processing ` until it

**Comment by hervenivon** [comment]
Thx @freddi-kit

**Comment by mathaeus** [comment]
> Ah, it looks take so long time even if connect side finished processing in my env, i think it can be skipped by skip_waiting_for_build_processing. Let me check later but I think it is issue of App side API because there are update of AppStore Connect API today.

@freddi-kit sounds like that problem could also be [this](https://github.com/fastlane/fastlane/issues/20645)

**Comment by freddi-kit** [comment]
ah! yes thank you!

**Comment by asincole** [comment]
> ```shell
> export FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1 
> ```

Hello, this may be a silly question but I had tried doing this before I saw the option of updating to the "altool" git branch (this did not work for me).

Now, though I have updated my gems to the "altool" branch and installed updates, fastlane still seems to attempt using transporter app (and fails). Is there a new value I should set "FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT" to or how do I unset it so fastlane uses FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT instead. 

Thanks for your help folks.

**Comment by mikehardy** [comment]
Thanks for all the work here - eager to see the final release with the aitool branch merged, hopefully after the API hanging thing was fixed. This looks like a frustrating little tangle of issues.

Posting what worked for me just now, since I use fastlane but without using bundler so installing from a git repo/branch is more difficult:

```
gem install specific_install
gem specific_install -l https://github.com/freddi-kit/fastlane.git -b altool-upload
```

Then I altered my `upload_to_testflight` to have parameter `skip_waiting_for_build_processing: true`, and :boom: it worked. Kudos!

**Comment by yolpsoftware** [comment]
> Dear all,
> 
> @freddi-kit PR #20631 looks good now...we have covered all the possible known scenarios. I am adding `status: workaround available` in this ticket because people are no longer blocked to start using Xcode14. Now Xcode14 users can switch the Fastlane version to PR #20631 in **local and CI** both and everything should work without any hack and without any change inside the Fastfile-scripts/pilot/deliver code.
> 
> ### Switch your Fastlane version to PR
> * Update `Gemfile` to 👇 and run `bundle install`, `bundle update fastlane`, or `bundle update`
> 
> ```ruby
> gem "fastlane", :git => "https://github.com/freddi-kit/fastlane.git", :branch => "altool-upload"
> ```
> 
> Please let us know if we are still missing any edge-case or something! To me, PR #20631 is simply fire 🔥🔥🔥 good to go 🚀 Thank you again @freddi-kit 👏

Thanks guys for this workaround. However, I'm not sure where I can find this `Gemfile`. My React Native project uses Expo Application Services (EAS) to build, which under the hood uses fastlane. I don't have a `Gemfile` anywhere in my project folder, and searching for `Gemfile` across the system reveals some, but they don't seem to be related to my project.

I tried placing a `Gemfile` with those contents in my project folder, but I'm getting the same error.

Any ideas?

**Comment by freddi-kit** [comment]
@yolpsoftware https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663 you may try it

**Comment by asincole** [comment]
> @asincole
> 
> sorry, I missed your comment. You don't need FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT and FASTLANE_ITUNES_TRANSPORTER_PATH if you use my branch. Or what error did you get?

hey @freddi-kit no worries, I can see you have your hands full. Thanks for all the work you are doing.

if I unset  FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT and FASTLANE_ITUNES_TRANSPORTER_PATH  then I get this when I try to push a beta with pilot

```
 [iTMSTransporter] An error (-10814) occurred. The operation couldn’t be completed. (OSStatus error -10814.) (null) #20650
Closed 
```

**Comment by yolpsoftware** [comment]
@freddi-kit thanks this worked indeed.

**Comment by freddi-kit** [comment]
> if I unset FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT and FASTLANE_ITUNES_TRANSPORTER_PATH then I get this when I try to push a beta with pilot


Please try to change Gemfile to use my branch https://github.com/fastlane/fastlane/issues/20371#issuecomment-1242768042

**Comment by asincole** [comment]
> > if I unset FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT and FASTLANE_ITUNES_TRANSPORTER_PATH then I get this when I try to push a beta with pilot
> 
> Please try to change Gemfile to use my branch [#20371 (comment)](https://github.com/fastlane/fastlane/issues/20371#issuecomment-1242768042) if you want to unset them

@freddi-kit oh yeah, I changed that too. this is what my gemfile looks like 

```
source "https://rubygems.org"

# gem "fastlane"
gem "fastlane", :git => "https://github.com/freddi-kit/fastlane.git", :branch => "altool-upload"
```

This is what my fastfile looks like 
```
platform :ios do
     # iOS Lanes
   desc 'Fetch certificates and provisioning profiles'
   lane :certificates do
     match(app_identifier: 'app id', type: 'development', readonly: true)
     match(app_identifier: 'app id', type: 'appstore', readonly: true)
   end

   desc 'Build the iOS application.'
   private_lane :build do
     certificates
     increment_build_number(xcodeproj: './App.xcodeproj')
     gym(workspace: './App.xcworkspace', clean: true)
   end

   desc 'Ship to Testflight.'
    lane :beta do
      api_key = app_store_connect_api_key(
  key_id: "#{ENV["FASTLANE_APP_STORE_CONNECT_KEY_ID"]}",
  issuer_id: "#{ENV["FASTLANE_APP_STORE_CONNECT_ISSUER_ID"]}",
  key_content: "{{key}}",
#         key_id: ENV['FASTLANE_APP_STORE_CONNECT_KEY_ID'],
#         issuer_id: ENV['FASTLANE_APP_STORE_CONNECT_ISSUER_ID'],
#         key_content: "#{ENV['FASTLANE_APP_STORE_CONNECT_KEY_CONTENT']}".gsub('\n', '\\n'),
#         key_content: ENV['FASTLANE_APP_STORE_CONNECT_KEY_CONTENT'],
        duration: 1200, # optional (maximum 1200)
        in_house: false # optional but may be required if using match/sigh
      )
      build
      pilot(
        api_key: api_key,
        skip_waiting_for_build_processing: true,
        distribute_external: true,
#         distribute_only: true,
        groups: [], # options[:groups]
        changelog: "This is not implemented yet"
      )
      clean_build_artifacts
#       commit_version_bump(message: 'Bump build', xcodeproj: './App.xcodeproj')
#       push_to_git_remote
    end
end
```

**Comment by freddi-kit** [comment]
@asincole 
hm, Did you run `bundle install`?

**Comment by asincole** [comment]
@freddi-kit yup, I did.

my gemfile is in the same parent folder as my fastlane folder. should the gemfile be in the fastlane folder?

**Comment by freddi-kit** [comment]
@asincole 

Let me check my code again. You also run fastlane by `bundle exec fastlane ios beta`, right?

**Comment by freddi-kit** [comment]
> should the gemfile be in the fastlane folder?

No, your current one is okay, I think

**Comment by asincole** [comment]
> bundle exec fastlane ios beta

@freddi-kit no, I run ```fastlane ios beta```

if I try to run ```bundle exec fastlane ios beta``` I get an error 


```
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios beta |
+------------------+----------+
[10:13:44]: invalid curve name

+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | default_platform       | 0           |
| 💥   | app_store_connect_api  | 0           |
|      | _key                   |             |
+------+------------------------+-------------+

[10:13:44]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️   invalid curve name after brew upgrade?
    https://github.com/fastlane/fastlane/issues/20553 [open] 7 💬
    a week ago

➡️  Getting error: invalid curve name for app_store_connect_api_key
    https://github.com/fastlane/fastlane/issues/20593 [open] 5 💬
    a week ago

➡️  Pilot fails using environment variables on GitHub
    https://github.com/fastlane/fastlane/issues/18655 [closed] 13 💬
    09 May 2022

and 5 more at: https://github.com/fastlane/fastlane/search?q=invalid%20curve%20name&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/usr/local/lib/ruby/gems/3.1.0/bin/fastlane)
/usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/spaceship/lib/spaceship/connect_api/token.rb:72:in `initialize': \e[31m[!] invalid curve name\e[0m (OpenSSL::PKey::ECError)
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/spaceship/lib/spaceship/connect_api/token.rb:72:in `new'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/spaceship/lib/spaceship/connect_api/token.rb:72:in `create'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/actions/app_store_connect_api_key.rb:43:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:44:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /usr/local/lib/ruby/gems/3.1.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /usr/local/lib/ruby/gems/3.1.0/bundler/gems/fastlane-cd2c504c7ba1/bin/fastlane:23:in `<top (required)>'
	from /usr/local/lib/ruby/gems/3.1.0/bin/fastlane:25:in `load'
	from /usr/local/lib/ruby/gems/3.1.0/bin/fastlane:25:in `<top (required)>'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `load'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `kernel_load'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:23:in `run'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:483:in `exec'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:31:in `dispatch'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:25:in `start'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.11/exe/bundle:48:in `block in <top (required)>'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/3.1.0/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /usr/local/Cellar/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.11/exe/bundle:36:in `<top (required)>'
	from /usr/local/opt/ruby/bin/bundle:25:in `load'
	from /usr/local/opt/ruby/bin/bundle:25:in `<main>'
```

**Comment by freddi-kit** [comment]
@asincole 
Okay, you need to run `bundle exec fastlane ios beta` if you install fastlane via `bundle install` with Gemfile. 

This error looks you specifying invalid `key_content` to `app_store_connect_api`, please check your key_content is correct forma.

**Comment by peagasilva** [comment]
@freddi-kit just to make sure: is your api key correct?

`invalid curve name` happened to me, but it turned out my key was incorrect - empty in my case.

I hope this can help pointing you to the right direction.

**Comment by asincole** [comment]
> @asincole Okay, you need to run `bundle exec fastlane ios beta` if you install fastlane via `bundle install` with Gemfile.
> 
> This error shows that you specifying invalid `key_content` to `app_store_connect_api`, please check your key_content is correct format.

@freddi-kit oh shit, sorry lol. I deleted my keys from the file because I wanted to paste them here. 

I've restored the keys and I'm rerunning the command. I unset those env variables and so far, I do not have a problem. I will update you when it finishes running if it works this time. thanks

**Comment by freddi-kit** [comment]
@asincole 
Good job 👍

**Comment by asincole** [comment]
@freddi-kit thanks a lot, it works fine now. I can't remember why I was getting an error yesterday. You are a lifesaver, thanks a lot. So moving forward, I always need to prefix with ```bundle exec``` before running my Fastlane command?

Same in CI yeah?

**Comment by freddi-kit** [comment]
> bundle exec before running my Fastlane command?

Yes, `bundle exec` loads the fastlane version which is specified in Gemfile. It helps to ensure behavior with other engineer's environment and trouble shooting and other things. Same with CI.

**Comment by asincole** [comment]
@freddi-kit thanks a lot. :-)

**Comment by freddi-kit** [comment]
@asincole 

Please call `bundle update fastlane` after my PR merged to fetch new version 😄

**Comment by freddi-kit** [comment]
Merged, let's wait new release version. I still wanna your Feedback because there may be missed case, thanks 🙇

**Comment by devon2018** [comment]
I've switched over to the master branch but still seem to be getting this error

`[!] Could not find transporter at /Applications/Xcode.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.`

my gemfile looks like this

```
source "https://rubygems.org"

gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "master"
```

and i ran bundle update fastlane and that worked correctly

**Comment by freddi-kit** [comment]
@devon2018 I think you may unset FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT or FASTLANE_ITUNES_TRANSPORTER_PATH

**Comment by freddi-kit** [comment]
Or can you rerun `bundle update fastlane` again? I looks there is new update about version

**Comment by freddi-kit** [comment]
@devon2018 it looks there are some same report from developer who tries to use master branch directly, please wait gem official update.

**Comment by freddi-kit** [comment]
Hello, if you got `[!] Could not find transporter at /Applications/Xcode.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.` error message, Please try below


- `bundle install` again with fastlane's master branch
-  unset FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT or FASTLANE_ITUNES_TRANSPORTER_PATH

If you cannot solve by above, please wait for new release of fastlane in RubyGems.

**Comment by freddi-kit** [comment]
@devon2018 please try official rubygem 2.210.0 version instead of master/freddi-kit branch

**Comment by freddi-kit** [comment]
@devon2018 could you share me latest logs?

**Comment by michaek** [comment]
I'm not sure if it is related to these changes, but it appears that `pilot` will (now?) hang without output after "Fetching password for transporter from environment variable named `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`" if you need to accept Xcode license agreements, while other fastlane commands will exit with a message informing the developer how to accept the agreements.

**Comment by freddi-kit** [comment]
@michaek I believe it is other issue and I cannot say whether it is related or not without any logs. Please open other issue

**Comment by freddi-kit** [comment]
@devon2018 How are you calling fastlane from CLI? `bundle exec fastlane` or `fastlane`?

**Comment by freddi-kit** [comment]
@devon2018 What will be shown if you type below?

```
bundle exec fastlane --version
```

**Comment by devon2018** [comment]
```
fastlane installation at path:
/Library/Ruby/Gems/2.6.0/bundler/gems/fastlane-5eaf30d538f6/bin/fastlane
-----------------------------
[✔] 🚀 
fastlane 2.210.0
```

**Comment by freddi-kit** [comment]
hm... it looks fastlane version is okay. Thanks

Is `bundle exec fastlane` still showing same issue? I'm happy if you paste error log.

**Comment by freddi-kit** [comment]
FYI: it looks brew one is not updated yet so please don't use brew one

**Comment by michaek** [comment]
@freddi-kit Since I have now accepted the license agreements, I don't know if I can produce the logs necessary to troubleshoot on a new issue. I only wanted to raise my observation in case someone else encountered it, or in case it rang an immediate "I see how that could happen" bell for maintainers. You don't need to take any action on this. 🙇

**Comment by devon2018** [comment]
@freddi-kit Yep managed to build and deploy using `bundle exec fastlane {your_lane}` not sure what changed between tries but all working now.

Thanks for the help :)

**Comment by freddi-kit** [comment]
@devon2018 oh, thank you too.

**Comment by freddi-kit** [comment]
@michaek oh, okay. I just checked now about fastlane with FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD, and i could not find any problem 🙇 But if you are in trouble, please tell me everything.

**Comment by freddi-kit** [comment]
@SumanDas94 deliver action is also okay

**Comment by SumanDas94** [comment]
Thank you. what is the fastlane version that has this change ?
Also quick question if we switch to the `altool` branch. do we need to export the above environment variables or can we directly go for pilot action ?

**Comment by freddi-kit** [comment]
@SumanDas94 
The new fastlane is released to use new altool implementation (2.210.0). And you don't need to set environment variables.
Please unset `FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT` and `FASTLANE_ITUNES_TRANSPORTER_PATH` if you want to try 2.210.0

**Comment by SumanDas94** [comment]
> The new fastlane is released to use new altool implementation (2.210.0). 
do i need to specify any branch for this ? or is it merged to the main branch ?
Also action remains as pilot ?

**Comment by freddi-kit** [comment]
> do i need to specify any branch for this ? or is it merged to the main branch ?

please specify gem in Gemfile normally like below. You may need to run `bundle update fastlane`

```
gem "fastlane"
```

Both pilot and deliver works

**Comment by SumanDas94** [comment]
Awesome let me try this

**Comment by alfavata** [comment]
Thanks for this @freddi-kit (and all the reviewers of the PR)

However even after successfully updating to 2.210.0 and unsetting the environment variables, fastlane still seems to be using Transporter for me (see log), which then fails as per the original issue - any idea? 😓 

>[15:24:06]: ----------------------------------
[15:24:06]: --- Step: upload_to_testflight ---
[15:24:06]: ----------------------------------
[15:24:06]: Creating authorization token for App Store Connect API
[15:24:07]: Ready to upload new build to TestFlight (App: 1553609146)...
[15:24:07]: Going to upload updated app to App Store Connect
[15:24:07]: This might take a few minutes. Please don't interrupt the script.
[15:24:07]: [iTMSTransporter] -Djava.ext.dirs=/Applications/Transporter.app/Contents/itms/java/lib/ext is not supported.  Use -classpath instead.
[15:24:07]: [iTMSTransporter] Error: Could not create the Java Virtual Machine.
[15:24:07]: [iTMSTransporter] Error: A fatal exception has occurred. Program will exit.
[15:24:07]: iTunes Transporter output above ^
[15:24:07]: The call to the iTMSTransporter completed with a non-zero exit status: 1. This indicates a failure.

**Comment by freddi-kit** [comment]
@alfavata You may still set FASTLANE_ITUNES_TRANSPORTER_PATH, please unset it

**Comment by alfavata** [comment]
> @alfavata But i feel your env still has `FASTLANE_ITUNES_TRANSPORTER_PATH`, could you check it by `echo $FASTLANE_ITUNES_TRANSPORTER_PATH` again?

It's definitely empty when I `echo` it 🤷

**Comment by freddi-kit** [comment]
hm...

**Comment by freddi-kit** [comment]
fastlane never loads Transporter.app if FASTLANE_ITUNES_TRANSPORTER_PATH is not set. so ... i would like you to do other way like opening new terminal.app and try. Could you try?

**Comment by freddi-kit** [comment]
You also can try to check by adding below code in Fastfile @alfavata 

```
puts ENV['FASTLANE_ITUNES_TRANSPORTER_PATH']
```

**Comment by alfavata** [comment]
Wen following your last suggestion I found the culprit 🤦 I had this in my Fastfile:
```
  before_all do
    ENV["FASTLANE_ITUNES_TRANSPORTER_PATH"]="/Applications/Transporter.app/Contents/itms"
  end
```

which I must've added months ago when first trying to solve the original issue!

**Comment by freddi-kit** [comment]
@alfavata 👍

**Comment by freddi-kit** [comment]
@ all who want to use 2.210.0

Please don't forget to unset environment variable which mentioned in https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663 

Also, please check your Gemfile version and what fastlane you are calling.

If you ask the question which related to them, I may not reply because I wanna focus on actual bug as possible 🙇

**Comment by SumanDas94** [comment]
@freddi-kit for this https://github.com/fastlane/fastlane/issues/20371#issuecomment-1248211699 unsetting is only required if we had earlier set it right to try out installed Transporter.app ?

**Comment by freddi-kit** [comment]
yes, and now you can uninstall Transporter.app

**Comment by SumanDas94** [comment]
I still have in installed , haven't removed it. Hoping that will be fine right for fastlane 210 version if i donot set the variables ?

**Comment by freddi-kit** [comment]
Again

> Please don't forget to unset environment variable which mentioned in https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663

**Comment by freddi-kit** [comment]
@SumanDas94 Ah, sorry. I got it. 🙇

**Comment by drewolbrich** [comment]
I updated to fastlane 2.210.0, removed the two `FASTLANE_ITUNES_TRANSPORTER*` environment variables, and was able to run `fastlane beta` and upload my app to TestFlight successfully.

Thank you, @freddi-kit 
❤️❤️❤️

**Comment by AlexScharch-BO** [comment]
First of all, thanks very much for putting this fix in place and addressing this issue so quickly, @freddi-kit! Really appreciate it. 🙂 

It looks like the changes do not make any updates to validation, as I just tried the latest Fastlane version (2.210.0) and validation still fails with the `Could not find transporter at /Applications/Xcode.app/Contents/Developer/.` error.

I also see this addition in `itunes_transporter.rb`:
```ruby
    def build_verify_command(username, password, source = "/tmp", provider_short_name = "", jwt = nil)
      raise "This feature has not been implemented yet with altool for Xcode 14"
    end
```

I'm wondering if there is any intention to expand this solution to fix validation as well. My team and I sometimes use App Store Connect validation to make sure that our CI deployment pipeline continues to function properly (without having to actually upload the .ipa file).

**Comment by freddi-kit** [comment]
@AlexScharch-BO Thank you for reporting it. And I'm sorry, in this PR I focused only update part and verify is not implemented, as `itunes_transporter.rb` says so 🙇 

If you want to use verify, please consider to use https://github.com/fastlane/fastlane/issues/20371#issuecomment-1239093663. I will work the verify part soon as possible so please wait a moment. 

To track issue easy, could you create new issue for it?

**Comment by AlexScharch-BO** [comment]
Hey @freddi-kit, I just created this issue: https://github.com/fastlane/fastlane/issues/20668

It's a bit bare-bones since my company is very security-minded, but hopefully it's sufficient since the issue here is already known. Please let me know if there's any more info I can provide.

I'll also say that getting App Store Connect validation fixed is not an immediate need for us. It's something that we will use from time to time when making updates to our CI/CD process, but it's not something that we use daily. Thus, I'm looking forward to the fix, but it is not urgent for us. Your fixing of the upload process here has already been a great help; thank you again for it!

**Comment by kami-mehar38** [comment]
> I found `FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT` to make success upload ipa to AppStore with Transporter.app
> 
> ```shell
> export FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT=1 
> export FASTLANE_ITUNES_TRANSPORTER_PATH=/Applications/Transporter.app/Contents/itms 
> bundle exec fastlane {your_action}
> ```
> 
> Please set FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT as 1 if you want to use Transporter.app.

Thanks, it worked 👍🏼

**Comment by drosenstark** [comment]
This worked with Xcode 14 production version and Fastlane 2.210.0.

One thing to note is that if you haven't updated your Fastlane in a long time, `bundle update fastlane` might not actually update your Fastlane. I don't know what random series of commands like `gem update` and `gem update bundler` and `bundle update fastlane` got me there, but.. make sure `fastlane --version` is reporting the version you actually need.

You don't need any other workarounds if you have the latest Fastlane. Thanks again.

**Comment by kami-mehar38** [comment]
> This worked with Xcode 14 production version and Fastlane 2.210.0.
> 
> 
> 
> One thing to note is that if you haven't updated your Fastlane in a long time, `bundle update fastlane` might not actually update your Fastlane. I don't know what random series of commands like `gem update` and `gem update bundler` and `bundle update fastlane` got me there, but.. make sure `fastlane --version` is reporting the version you actually need.
> 
> 
> 
> You don't need any other workarounds if you have the latest Fastlane. Thanks again.
> 
> 
> 
> Edit: Apparently with verify-only updating Fastlane isn't enough. So there's that ;)

Xcode version 14 is confirmed but probably I haven't upgraded fasltane for quite a while. Anyway I have already tried the commands mentioned and it worked. 

I shall have a look at fastlane version too. 

Thanks 👍🏻

**Comment by freddi-kit** [comment]
@SumanDas94 

> Can understand it is being thrown from altool.

yep

>  Is there a limit on the number of TF builds we can upload via pilot ?

I don't detail about it, please investigate by yourself 🙇

**Comment by niyaoyao** [comment]
> ITMSTransporter

It' worked for me with Xcode 14.0 (Build version 14A309)

**Comment by willokyes** [comment]
> > ITMSTransporter
> 
> It's worked for me with Xcode 14.0 (Build version 14A309).
> 
> ```ruby
> before_all do |_lane, options|
>   ENV["FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT"]="1"
>   ENV["FASTLANE_ITUNES_TRANSPORTER_PATH"]="/Applications/Transporter.app/Contents/itms"
> end
> ```

it's also work for me ! thank you !

**Comment by freddi-kit** [comment]
@willokyes @niyaoyao I'm not sure why you are still using old fastlane, but you don't need to do them in newest fastlane version with Xcode 14

**Comment by sheng5566** [comment]
> > ITMSTransporter
> 
> It's worked for me with Xcode 14.0 (Build version 14A309).
> 
> ```ruby
> before_all do |_lane, options|
>   ENV["FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT"]="1"
>   ENV["FASTLANE_ITUNES_TRANSPORTER_PATH"]="/Applications/Transporter.app/Contents/itms"
> end
> ```

put this code on top of the fastfile or?

**Comment by willokyes** [comment]
> > > ITMSTransporter
> > 
> > 
> > It's worked for me with Xcode 14.0 (Build version 14A309).
> > ```ruby
> > before_all do |_lane, options|
> >   ENV["FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT"]="1"
> >   ENV["FASTLANE_ITUNES_TRANSPORTER_PATH"]="/Applications/Transporter.app/Contents/itms"
> > end
> > ```
> 
> put this code on top of the fastfile or?

YES ！

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/25343908/197172817-48191725-5982-4d4e-879a-4e17b02d8de2.png">

**Comment by stherold** [comment]
> Can understand it is being thrown from altool. Is there a limit on the number of TF builds we can upload via pilot ?

How many apps do you upload? We might encounter the same issue one day...

**Comment by stherold** [comment]
@freddi-kit we're now getting this wonderful error: 

```
ERROR [2022-10-21 15:57:34.61]: Error uploading '/var/folders/9l/1ysg9vq51p37q0lgksxhdb9r0000gn/T/d20221021-14150-cqk0nc/1400408720-d278185c-82a7-485e-8276-fbcc520610c3.itmsp'.
ERROR [2022-10-21 15:57:34.61]: Could not determine the package’s bundle ID. The package is missing an Info.plist or the CFBundlePackageType is not ‘APPL’ or ‘FMWK’. Unable to validate your application. (-21017)
ERROR [2022-10-21 15:57:34.61]: The call to the altool completed with a non-zero exit status: 1. This indicates a failure.
ERROR [2022-10-21 15:57:34.61]: Could not download/upload from App Store Connect!
```

I don't know why `altool` wants to upload an `itmsp` file. Any idea that pops up? We definitely have an IPA file as the lane context shows:

```
{
    :ENVIRONMENT=>"<redacted>.", 
    :DEFAULT_PLATFORM=>:ios,
    :PLATFORM_NAME=>:ios,
    :LANE_NAME=>"ios beta", 
    :VERSION_NUMBER=>"<redacted>.", 
    :XCODEBUILD_DERIVED_DATA_PATH=>nil, 
    :SCAN_GENERATED_XCRESULT_PATH=>"/Users/vagrant/Library/Developer/Xcode/DerivedData/<redacted>.-fkftxgyqzutwhfgakscnfpopktat/Logs/Test/Test-<redacted>.-2022.10.21_15-46-58-+0000.xcresult", 
    :SCAN_DERIVED_DATA_PATH=>"/Users/vagrant/Library/Developer/Xcode/DerivedData/<redacted>.-fkftxgyqzutwhfgakscnfpopktat", 
    :SCAN_GENERATED_PLIST_FILES=>[], 
    :SCAN_GENERATED_PLIST_FILE=>nil, 
    :BUILD_NUMBER=>"105105", 
    :IPA_OUTPUT_PATH=>"/Users/vagrant/deploy/<redacted>..ipa", 
    :XCODEBUILD_ARCHIVE=>"/Users/vagrant/Library/Developer/Xcode/Archives/2022-10-21/<redacted>. 2022-10-21 15.54.20.xcarchive", 
    :DSYM_OUTPUT_PATH=>"/Users/vagrant/deploy/<redacted>.app.dSYM.zip"
}
```

**Comment by 0xpablo** [comment]
Hi @stherold, we were having the same issue, by looking at Fastlane's sources, I saw that if you have the `ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD` environment variable set, even if Fastlane uses `altool` it affects the resulting path as you can see [here](https://github.com/fastlane/fastlane/blob/9db0159cba89885c5d760294e8c6c7e696bba031/fastlane_core/lib/fastlane_core/itunes_transporter.rb#L745).

After removing that env var, `altool` didn't complain anymore.

**Comment by LugonjaAleksandar** [comment]
Did anyone figure out how to setup specific version (2.210.0) on github actions? There is no option provided for it, and the error persists for `maierj/fastlane-action@v2.3.0`

**Comment by aoscodes** [comment]
> Did anyone figure out how to setup specific version (2.210.0) on github actions? There is no option provided for it, and the error persists for `maierj/fastlane-action@v2.3.0`

@LugonjaAleksandar it might be worth just using fastlane in GH Actions (rather than that custom action, which appears to be just be providing an api over fastlane)

fastlane comes pre-installed on all GH Action runners. Presuming you have the correct version of fastlane in your gems then you should just be able to run `bundle exec fastlane my_cool_lane`

**Comment by gancoce94** [comment]
Tks @freddi-kit love you bro, is working fine! 🍻

**Comment by lnanhkhoa** [comment]
you save my life, bro @freddi-kit

**Comment by leopthurler** [comment]
> This worked with Xcode 14 production version and Fastlane 2.210.0.
> 
> One thing to note is that if you haven't updated your Fastlane in a long time, `bundle update fastlane` might not actually update your Fastlane. I don't know what random series of commands like `gem update` and `gem update bundler` and `bundle update fastlane` got me there, but.. make sure `fastlane --version` is reporting the version you actually need.
> 
> You don't need any other workarounds if you have the latest Fastlane. Thanks again.
> 
> Edit: Apparently with verify-only updating Fastlane isn't enough. So there's that ;)

This solution worked for me.

**Comment by Shayokh144** [comment]
Updating to **latest** **Fastlane** **version** fixed this problem for me.

**Comment by useful-links** [comment]
can anyone provide an example of how to implement upload to TestFlight with the old method using "upload_to_testflight" ,
and what needs to be done now?

old solution (Xcode < 14):
`
api_key = app_store_connect_api_key(
    key_id: "#{key_id}",
    issuer_id: "#{issuer_id}",
    key_filepath: "#{key}",
    duration: 1200,
    in_house: false
  )
  
upload_to_testflight(
    api_key: "#{api_key}",
    app_identifier: "#{app_identifier}",
    changelog: change_log,
    ipa: ipa_path,
    uses_non_exempt_encryption: true
  )
`
what does it look like now ?

**Comment by useful-links** [comment]
now i upload like this:

app_store_connect_api_key(
key_id: "#{key_id}",
issuer_id: "#{issuer_id}",
key_filepath: "#{key}",
duration: 1200,
in_house: false
)

xcrun altool --list-providers --apiKey #{key_id} --apiIssuer #{issuer_id}

xcrun altool --validate-app -f #{ipa_path} -t ios --apiKey #{key_id]} --apiIssuer #{issuer_id}

**Comment by pyby** [comment]
Since this issue is close, with Fastlane 2.210.0 release:

```
gym(
  [… your optional options]
)

app_store_connect_api_key(
  key_id: #{key_id}",
  issuer_id: #{issuer_id},
  key_filepath: #{key_file_path}
)

# upload_to_testflight, alias of pilot
pilot(
  [… your optional options]
)
```

Nothing to do more.

**Comment by frederikbosch** [comment]
If anyone hits 

> ERROR ITMS-90725: SDK version issue. This app was built with the iOS 15.5 SDK. All iOS and iPadOS apps submitted to the App Store must be built with the iOS 16.1 SDK or later, included in Xcode 14.1 or later

the solution is to stop using `iTMSTransporter` and switch to `altool`.

**Comment by ykhandelwal913** [comment]
I am seeing the same issue with Xocde14.2 and Xcode14.3 with `  lane :push_to_testflight do |options|
        #Variable required to be set for 2FA enabled accounts: FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD
        testflight(
            username: ENV['APPLE_ACCOUNT_USER'],
            skip_waiting_for_build_processing: true,
            apple_id: ENV['APP_APPLE_ID'],
            team_name: "XYZ",
            ipa: options[:ipa_location]
        )
    end #`

`Could not find transporter at /Applications/Xcode14.2.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.


+------+------------------+-------------+

|           fastlane summary            |

+------+------------------+-------------+

| Step | Action           | Time (in s) |

+------+------------------+-------------+

| 1    | xcode_select     | 0           |

| 2    | default_platform | 0           |

| 3    | is_ci            | 0           |

| 4    | setup_jenkins    | 0           |

| 💥   | testflight       | 0           |

+------+------------------+-------------+


[04:44:17]: fastlane finished with errors


[!] Could not find transporter at /Applications/Xcode14.2.app/Contents/Developer/. Please make sure you set the correct path to your Xcode installation.
`

Can someone help how can i migrate to altool for lane action `testflight`

**Comment by ykhandelwal913** [comment]
I am still seeing this issue with xcode14.3 using `pilot`

**Comment by dhivya0413** [comment]
I am still seeing this issue with xcode14.2 using pilot

**Comment by Dat-Mobile** [comment]
Remove all the values previously added, include:
```
ENV["ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD"]
ENV["FASTLANE_ITUNES_TRANSPORTER_USE_SHELL_SCRIPT"]
ENV["FASTLANE_ITUNES_TRANSPORTER_PATH"]
```
and it works fine now

**Comment by ykhandelwal913** [comment]
@Dat-Mobile can you share the fastfile and pilot configuration?

**Comment by Dat-Mobile** [comment]
```
fastlane (2.212.2)
upload_to_testflight(
      apple_id: appId,
      skip_waiting_for_build_processing: true
 )
```
this is all from me

**Comment by ykhandelwal913** [comment]
Thanks @Dat-Mobile and your xcode version is 14.3..right?

**Comment by Dat-Mobile** [comment]
No, it's 14.2

**Comment by corentin35000** [comment]
Again the error: An error (-10814) occurred. The operation couldn’t be completed. (OSStatus error -10814.) (null)
, with Xcode version 15.4.0 and also 16.0.0 (beta). I find it really surprising, altool is deprecated, but since version 14.0 iTMSTransporter does not work! Two major versions after that it's still the same thing.. the images of the macos runners from github actions start from 14.3.1 so we can't get around with Xcode 13.x.x, anyone have a solution?

**Comment by freddi-kit** [comment]
@corentin35000

I'm not sure why it happens on 15.4, but about 16.0, Apple may not accept beta version fro AppStore Connect

**Comment by freddi-kit** [comment]
> altool is deprecated

Could you share me the source of this information?

**Comment by freddi-kit** [comment]
@corentin35000 please confirm that your environment shows result of `xcrun altool`
